### PR TITLE
Javaparser for  convert scr doc comments to osgi annotations

### PR DESCRIPTION
**Pull request by DinethUWU**
## Purpose
Project for refactoring scr doclets (doc comments) to SCR Java annotations and add those things to component file as osgi annotations.

### CODE REVIEW

File: wso2.org.component.javaparser/.idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        <sourceTestOutputDir name="target/generated-test-sources/test-annotations" />
+        <outputRelativeToContentRoot value="true" />
+        <module name="wso2orgcomponentjavaparser" />
+      </profile>
+    </annotationProcessing>
+    <bytecodeTargetLevel>
+      <module name="wso2orgcomponentjavaparser" target="1.8" />
+    </bytecodeTargetLevel>
+  </component>
+</project>
```
Comment (kasunbg) [inline wso2.org.component.javaparser/.idea/compiler.xml]: plz remove the IDE specific files.

File: wso2.org.component.javaparser/.idea/encodings.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="Encoding">
+    <file url="file://$PROJECT_DIR$" charset="UTF-8" />
+  </component>
+</project>
```
Comment (kasunbg) [inline wso2.org.component.javaparser/.idea/encodings.xml]: .

File: wso2.org.component.javaparser/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+This project for refactoring scr doc comment from declarative service file and and add those things to same file as osgi annotations.
```
Comment (kasunbg) [inline wso2.org.component.javaparser/README.md]: * from scr doclets (doc comments) to SCR Java annotations.

File: wso2.org.component.javaparser/converter.sh [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+git clone --depth 1 https://github.com/$REPO.git
+
+
+JAVA_PARSER=wso2.org.component.javaparser-1.0.0-SNAPSHOT-jar-with-dependencies.jar
+
+LOCAL_REPO=$(ls -td -- */ | head -n 1)
+
+
+
+repo_result=$(find . -name repo_result.txt)
```
Comment (kasunbg) [inline wso2.org.component.javaparser/converter.sh]: so many spaces.. :astonished:

File: wso2.org.component.javaparser/converter.sh [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+ else
+ echo "The Repository name is incorrect. It should look like: wso2/product-is"
+ fi
+done
+
+
+
+
+
+
```
Comment (kasunbg) [inline wso2.org.component.javaparser/converter.sh]: .

File: wso2.org.component.javaparser/pom.xml [hunk 6]
```diff
@@ -0,0 +1,109 @@
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>wso2.org</groupId>
```
Comment (kasunbg) [inline wso2.org.component.javaparser/pom.xml]: plz change this to org.wso2.build

File: wso2.org.component.javaparser/pom.xml [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    <artifactId>wso2.org.component.javaparser</artifactId>
```
Comment (kasunbg) [inline wso2.org.component.javaparser/pom.xml]: org.wso2.build.scr-annotation-converter

File: wso2.org.component.javaparser/src/main/java/component/javaparser/App.java [hunk 8]
```diff
@@ -0,0 +1,139 @@
+package component.javaparser;
+
```
Comment (kasunbg) [inline wso2.org.component.javaparser/src/main/java/component/javaparser/App.java:17]: add license.

File: wso2.org.component.javaparser/src/main/java/component/javaparser/CompareXMLDiff.java [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+public class CompareXMLDiff {
```
Comment (kasunbg) [inline wso2.org.component.javaparser/src/main/java/component/javaparser/CompareXMLDiff.java]: missing class comments..

File: wso2.org.component.javaparser/src/main/java/component/javaparser/WSO2JavaParser.java [hunk 10]
```diff
@@ -0,0 +1,265 @@
...
+        } catch (ParserConfigurationException e) {
+            e.printStackTrace();
+        }
+        Document doc = null;
+        try {
+            doc = dBuilder.parse(inputFile);
+        } catch (SAXException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (kasunbg) [inline wso2.org.component.javaparser/src/main/java/component/javaparser/WSO2JavaParser.java]: did we use a logging mechanism? I do not think we need one for this though.

File: wso2.org.component.javaparser/src/main/java/component/javaparser/WSO2POMReader.java [hunk 11]
```diff
@@ -0,0 +1,291 @@
...
+import javax.xml.transform.stream.StreamResult;
+
+/**
+ * This class read and process the pom files of repository
+ */
+public class WSO2POMReader {
+
+    private void addSCRPluginVersion(Document doc, NodeList nodeList) {
+        Element ele = null;
+        String version = "1.16.0";
```
Comment (kasunbg) [inline wso2.org.component.javaparser/src/main/java/component/javaparser/WSO2POMReader.java]: Better if we can make this version an input parameter. At least make this a constant.. possible name: DEFAULT_SCR_PLUGIN_VERSION.

File: wso2.org.component.javaparser/src/resources/ApplicationManagementServiceComponent.java [hunk 12]
```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (kasunbg) [inline wso2.org.component.javaparser/src/resources/ApplicationManagementServiceComponent.java]: test files need to go to src/test/resources/

File: wso2.org.component.javaparser/wso2.org.component.javaparser.iml [hunk 13]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module org.jetbrains.idea.maven.project.MavenProjectsManager.isMavenModule="true" type="JAVA_MODULE" version="4">
```
Comment (kasunbg) [inline wso2.org.component.javaparser/wso2.org.component.javaparser.iml]: remove.

File: wso2.org.component.javaparser/src/test/java/component/javaparser/AppTest.java [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+import junit.framework.Test;
+import junit.framework.TestCase;
+import junit.framework.TestSuite;
+
+/**
+ * Unit test for simple App.
+ */
+public class AppTest
+        extends TestCase {
+    /**
```
Comment (kasunbg) [inline wso2.org.component.javaparser/src/test/java/component/javaparser/AppTest.java]: remove.

File: wso2.org.component.javaparser/.idea/compiler.xml (hunk 1 above)
Comment (DinethUWU) [reply wso2.org.component.javaparser/.idea/compiler.xml]: removed files

File: wso2.org.component.javaparser/.idea/encodings.xml (hunk 2 above)
Comment (DinethUWU) [reply wso2.org.component.javaparser/.idea/encodings.xml]: removed

File: wso2.org.component.javaparser/README.md (hunk 3 above)
Comment (DinethUWU) [reply wso2.org.component.javaparser/README.md]: changed the content

File: wso2.org.component.javaparser/converter.sh (hunk 4 above)
Comment (DinethUWU) [reply wso2.org.component.javaparser/converter.sh]: removed spaces.

File: wso2.org.component.javaparser/pom.xml (hunk 6 above)
Comment (DinethUWU) [reply wso2.org.component.javaparser/pom.xml]: changed.

File: wso2.org.component.javaparser/pom.xml (hunk 7 above)
Comment (DinethUWU) [reply wso2.org.component.javaparser/pom.xml]: changed.

File: wso2.org.component.javaparser/src/main/java/component/javaparser/App.java (hunk 8 above)
Comment (DinethUWU) [reply wso2.org.component.javaparser/src/main/java/component/javaparser/App.java:17]: added license.

File: wso2.org.component.javaparser/src/main/java/component/javaparser/CompareXMLDiff.java (hunk 9 above)
Comment (DinethUWU) [reply wso2.org.component.javaparser/src/main/java/component/javaparser/CompareXMLDiff.java]: added class comments.

File: wso2.org.component.javaparser/src/main/java/component/javaparser/WSO2JavaParser.java (hunk 10 above)
Comment (DinethUWU) [reply wso2.org.component.javaparser/src/main/java/component/javaparser/WSO2JavaParser.java]: No. I removed this.

File: wso2.org.component.javaparser/src/main/java/component/javaparser/WSO2POMReader.java (hunk 11 above)
Comment (DinethUWU) [reply wso2.org.component.javaparser/src/main/java/component/javaparser/WSO2POMReader.java]: fixed this one.

File: wso2.org.component.javaparser/src/resources/ApplicationManagementServiceComponent.java (hunk 12 above)
Comment (DinethUWU) [reply wso2.org.component.javaparser/src/resources/ApplicationManagementServiceComponent.java]: added to src/test/resources/

File: wso2.org.component.javaparser/src/test/java/component/javaparser/AppTest.java (hunk 14 above)
Comment (DinethUWU) [reply wso2.org.component.javaparser/src/test/java/component/javaparser/AppTest.java]: removed.

File: wso2.org.component.javaparser/wso2.org.component.javaparser.iml (hunk 13 above)
Comment (DinethUWU) [reply wso2.org.component.javaparser/wso2.org.component.javaparser.iml]: removed.

File: wso2.org.component.javaparser/src/main/java/component/javaparser/CompareXMLDiff.java [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+    private int numberOfReferenceTag;
+    private List<String> fileDifferences = new ArrayList<>();
+
+    public void compareXML(String file1, String file2) throws ParserConfigurationException, IOException, SAXException {
+        DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+        DocumentBuilder db = null;
+        Document doc = null;
+        Document doc2 = null;
+            db = dbf.newDocumentBuilder();
+            doc = db.parse(new File(file1));
```
Comment (kasunbg) [inline wso2.org.component.javaparser/src/main/java/component/javaparser/CompareXMLDiff.java]: Please format the above code. Checkstyle plugin can be run from IDE against the modified source files to find formatting issues in your code.

File: wso2.org.component.javaparser/src/main/java/component/javaparser/WSO2POMReader.java [hunk 16]
```diff
@@ -0,0 +1,293 @@
...
+            dependecy.appendChild(artifactId);
+            dependecy.appendChild(version);
+        }
+    }
+
+    public void updatePomFile(Document doc, String file) throws TransformerException, IOException {
+        doc.setXmlStandalone(true);
+        doc.getDocumentElement().normalize();
+        TransformerFactory transformerFactory = TransformerFactory.newInstance();
+            Transformer transformer = transformerFactory.newTransformer();
```
Comment (kasunbg) [inline wso2.org.component.javaparser/src/main/java/component/javaparser/WSO2POMReader.java]: Please format the above code. Checkstyle plugin can be run from IDE against the modified source files to find formatting issues in your code.

**Review by kasunbg — APPROVED** [review]

File: wso2.org.component.javaparser/src/main/java/component/javaparser/WSO2POMReader.java (hunk 16 above)
Comment (DinethUWU) [reply wso2.org.component.javaparser/src/main/java/component/javaparser/WSO2POMReader.java]: fixed checkstyle.

File: wso2.org.component.javaparser/src/main/java/component/javaparser/CompareXMLDiff.java (hunk 15 above)
Comment (DinethUWU) [reply wso2.org.component.javaparser/src/main/java/component/javaparser/CompareXMLDiff.java]: fixed checkstyle.

**Comment by maheshika** [comment]
staled

**Pull request closed without merging**